src/core/net/SRP_CODER.md

```diff
@@ -0,0 +1,284 @@
+- Segments are one byte (8-bit) long except for the first segment which may have fewer bits (2-7 bits).
+- First (MSB) bit in each segment is the “continuation bit” indicating whether there are more segments to follow (`1`) or if it is the last segment (`0`).
+- The remaining bits after the MSB provide the numerical `uint` bit values in big-endian order.
+
+### Examples
+
+- Numbers less than (2^7) 128 are encoded as a single byte.
+- Numbers between 128 and (2^14 - 1) 16,383 are encoded as two bytes.
+- Numbers between 16,384 up to (2^21 - 1) 2,097,151 are encoded as three bytes.
+
+## DNS Name and Label Format
+
+### Label Dispatch
+
+```
+    0   1   2   3   4   5   6   7
+  +---+---+---+---+---+---+---+---+
+  | Type  ... | ...               |
+  +---+---+---+---+---+---+---+---+
+
+```
+
+- `00 <6-bit len>`: Standard label encoding, where the lower 6 bits indicate the label's length (number of characters). This is followed by the label characters, totaling `len` in number.
+- `01 <6-bit len>`: Similar to the above, but the label begins with an underscore `'_'`. The `len` value here doesn't include the underscore itself, which isn't explicitly encoded after the dispatch byte.
+- `10 <6-bit offset first seg>`: Instead of encoding a new label, this references a previously encoded label within the message. The 6-bit value in the dispatch byte serves as the first segment of a compact `uint` representing the offset to that label. This offset must point to a previous label within the message. Note that this only applies to a single label; any subsequent labels are encoded immediately after the current one.
+- `110 <5-bit code>`: Commonly used constant labels. Codes are:
+  - `0`: `_udp`
+  - `1`: `_tcp`
+  - `2`: `_matter`
+  - `3`: `_matterc`
+  - `4`: `_matterd`
+  - `5`: `_hap`
+- `111 <5-bit code>`: Commonly used generative patterns. Codes are:
+  - `0`: Single 64-bit value converted to a 16-char hexadecimal string using capital letters. The 64-bit value is encoded in binary format (big-endian). This pattern is used for target host name in matter.
+  - `1`: Two 64-bit values each expressed as a 16-char hexadecimal string, using capital letters, separated by a hyphen `-`. The two 64-bit values are encoded in binary format (8 bytes big-endian order). This pattern is used for service instance name in matter.
+  - `2`: Subtype label as `_<char><hhhh>` where a single char is encoded first, followed by 8 byte encoded 64-bit value for `<hhhh>` which is converted to a 16-char hexadecimal string, using capital letters.
+  - `3` : Subtype label as `_<char><hhhh>`, similar to previous case but instead of encoding the 8-byte value directly, an offset to an earlier instance of the same byte sequence is encoded in the message. The offset is encoded using compact `uint` format.
+
+A DNS name is encoded as a sequence of labels, ending with an empty label encoded as a single `0x00` byte. DNS names do not include the domain name.
```

Comment:
Note that a pointer ('10') can also terminate a domain name, and it might be worth mentioning that, since otherwise the implementor might put a null byte after the pointer.

Reply:
Updated in the new pushed commit.